src/resolvers/Organization/posts.ts

```diff
@@ -0,0 +1,133 @@
+ *
+ * @returns A Promise that resolves to an object containing an array of posts, the total count of posts, and pagination information. The pagination information includes the `startCursor`, `endCursor`, `hasPreviousPage`, and `hasNextPage`.
+ *
+ * @throws GraphQLError Throws an error if the provided arguments are invalid.
+ */
+export const posts: OrganizationResolvers["posts"] = async (parent, args) => {
+  const parseGraphQLConnectionArgumentsResult =
+    await parseGraphQLConnectionArguments({
+      args,
+      parseCursor: (args) =>
+        parseCursor({
+          ...args,
+          organization: parent._id,
+        }),
+      maximumLimit: MAXIMUM_FETCH_LIMIT,
+    });
+  if (!parseGraphQLConnectionArgumentsResult.isSuccessful) {
+    throw new GraphQLError("Invalid arguments provided.", {
+      extensions: {
+        code: "INVALID_ARGUMENTS",
+        errors: parseGraphQLConnectionArgumentsResult.errors,
+      },
+    });
+  }
+  const { parsedArgs } = parseGraphQLConnectionArgumentsResult;
+
+  const filter = getCommonGraphQLConnectionFilter({
+    cursor: parsedArgs.cursor,
+    direction: parsedArgs.direction,
+  });
+
+  //get the sorting object
+
+  const sort = getCommonGraphQLConnectionSort({
+    direction: parsedArgs.direction,
+  });
+
+  const [objectList, totalCount] = await Promise.all([
+    Post.find({
+      ...filter,
+      organization: parent._id,
+    })
+      .sort(sort)
+      .limit(parsedArgs.limit)
+      .lean()
+      .exec(),
+
+    Post.find({
+      organization: parent._id,
+    })
+      .countDocuments()
+      .exec(),
+  ]);
+
+  return transformToDefaultGraphQLConnection<
+    ParsedCursor,
+    InterfacePost,
+    InterfacePost
+  >({
+    objectList,
+    parsedArgs,
+    totalCount,
+  });
+};
+/*
+This is typescript type of the parsed cursor for this connection resolver.
+*/
+type ParsedCursor = string;
+
+/**
+ * This function is used to validate and transform the cursor passed to the `posts` connection resolver.
+ *
+ * @param args - An object that includes the cursor value, cursor name, cursor path, and the ID of the creator.
+ *
+ * @returns A Promise that resolves to an object that includes a boolean indicating whether the operation was successful, and the parsed cursor value. If the operation was not successful, the object also includes an array of errors.
+ *
+ * @throws Error Throws an error if the provided cursor is invalid.
+ * */
+export const parseCursor = async ({
+  cursorValue,
+  cursorName,
+  cursorPath,
+  organization,
+}: ParseGraphQLConnectionCursorArguments & {
+  organization: string | Types.ObjectId;
+}): ParseGraphQLConnectionCursorResult<ParsedCursor> => {
+  const errors: DefaultGraphQLArgumentError[] = [];
+  const post = await Post.findOne({
+    _id: cursorValue,
+    organization,
+  });
+  if (!post) {
+    errors.push({
+      message: `Argument ${cursorName} is an invalid cursor.`,
+      path: cursorPath,
+    });
+    if (errors.length !== 0) {
```

Comment:
get this if statement out of the previous if block, it should exist as a seperate check, do it like it is done in childTags.ts

Reply:
done